### ISSUE TITLE
Make use of sitelinks from Wikidata

### ISSUE THREAD

**Issue by lydiapintscher** [Jk40git/Wikidata-OCI-Extension#5]
Right now there is one link to Wikipedia at the top of the extension. That goes to a generic page that matches the string searched for if I understand it correctly.
It might be nicer to instead add one Wikipedia link to each result entry in the list that goes directly to the right article. You could use the sitelinks on the Item for this. Example: https://www.wikidata.org/wiki/Q64#sitelinks-wikipedia
Here is a sketch of what this could look like:
![IMG20240918104854](https://github.com/user-attachments/assets/7b29c615-bec1-421e-8141-006bfbf18598)

**Comment by Jk40git** [comment]
Hi @lydiapintscher closing this issue now as it's been solved we put the sitelinks on each result as a clickable element to send the user to the corresponding pages. Thanks for the idea

**Comment by YoshiRulz** [comment]
https://github.com/Jk40git/Wikidata-OCI-Extension/commit/d4625a7baaacec32310b6e014d204a01448185cd You did not use sitelinks, you did a string concatenation:
https://github.com/Jk40git/Wikidata-OCI-Extension/blob/21afe0b371a096c3afbf24639e294a959d41d00a/Popup/popup.js#L682
That won't work if the label on Wikidata doesn't match the article address.
- When multiple Items have the same label, at most 1 will match the address. Try it on `Berlin`.
- And it doesn't account for [MediaWiki special characters](https://en.wikipedia.org/wiki/WP:NCTR). Apparently there's an album called `~~~`, but try the extension on that and you'll get taken to an error page.

If the API you're using doesn't include sitelinks, you can just use e.g. `https://www.wikidata.org/wiki/Special:GoToLinkedPage/enwiki/Q184754`.

**Comment by Jk40git** [comment]
Thanks @YoshiRulz for your contribution. Can you explain what's a sitelink ? 
Can you show us how you get the error and the error page ?
And if you don't mind committing a patch I will be glad to discuss it there.

**Comment by YoshiRulz** [comment]
https://www.wikidata.org/wiki/Help:Sitelinks

To get to [this error page](https://en.wikipedia.org/wiki/~~~), I selected `~~~`, triggered the extension, then clicked the second result "2015 EP by Ana Roxanne".

I looked up the `wbsearchentities` API and it doesn't seem to include sitelinks, so my only idea would be to use `Special:GoToLinkedPage`. But that gives an error if an item doesn't have a sitelink in the chosen language(s). edit: Forgot about the Hub redirect service! PR incoming...
